# CCTV & 인구 데이터 합치기

In [4]:
import pandas as pd

CCTV_Seoul = pd.read_csv("./result_data/02_CCTV_result.csv")
popul_seoul = pd.read_excel("./result_data/02_Popul_result.xlsx")

In [5]:
CCTV_Seoul.head(1)

구별    소계  2013년도 이전  2014년  2015년  2016년       최근증가율
0  강남구  3238       1292    430    584    932  150.619195

In [6]:
popul_seoul.head(1)

구별     인구수     한국인   외국인    고령자     외국인비율      고령자비율
0  종로구  164257  154770  9487  26182  5.775705  15.939656

#### 🔰 두 데이터 병합

In [8]:
merge_data = pd.merge(CCTV_Seoul, popul_seoul, on="구별")
merge_data.head()

구별    소계  2013년도 이전  2014년  2015년  2016년       최근증가율     인구수     한국인  \
0  강남구  3238       1292    430    584    932  150.619195  561052  556164   
1  강동구  1010        379     99    155    377  166.490765  440359  436223   
2  강북구   831        369    120    138    204  125.203252  328002  324479   
3  강서구   911        388    258    184     81  134.793814  608255  601691   
4  관악구  2109        846    260    390    613  149.290780  520929  503297   

     외국인    고령자     외국인비율      고령자비율  
0   4888  65060  0.871220  11.596073  
1   4136  56161  0.939234  12.753458  
2   3523  56530  1.074079  17.234651  
3   6564  76032  1.079153  12.500021  
4  17632  70046  3.384722  13.446362

#### 🔰 필요 없는 연도별 데이터 컬럼 삭제

- del
- drop()

In [9]:
del merge_data["2013년도 이전"]
del merge_data["2014년"]
merge_data.head(3)

구별    소계  2015년  2016년       최근증가율     인구수     한국인   외국인    고령자     외국인비율  \
0  강남구  3238    584    932  150.619195  561052  556164  4888  65060  0.871220   
1  강동구  1010    155    377  166.490765  440359  436223  4136  56161  0.939234   
2  강북구   831    138    204  125.203252  328002  324479  3523  56530  1.074079   

       고령자비율  
0  11.596073  
1  12.753458  
2  17.234651

In [10]:
merge_data.drop(["2015년", "2016년"], axis=1, inplace=True)
merge_data.head()

구별    소계       최근증가율     인구수     한국인    외국인    고령자     외국인비율      고령자비율
0  강남구  3238  150.619195  561052  556164   4888  65060  0.871220  11.596073
1  강동구  1010  166.490765  440359  436223   4136  56161  0.939234  12.753458
2  강북구   831  125.203252  328002  324479   3523  56530  1.074079  17.234651
3  강서구   911  134.793814  608255  601691   6564  76032  1.079153  12.500021
4  관악구  2109  149.290780  520929  503297  17632  70046  3.384722  13.446362

#### 🔰 Index 지정

- 시각화 작업을 위해 인덱스를 변경한다.
- 인덱스로 지정하려면 중복되는 값이 없어야 한다.
- 인덱스를 재정의 하는 메서드는 `set_index()`이다.
- `set_index(__column)`는 선택한 컬럼을 DataFrame의 인덱스로 지정한다.

In [11]:
merge_data.set_index("구별", inplace=True)
merge_data.head()

소계       최근증가율     인구수     한국인    외국인    고령자     외국인비율      고령자비율
구별                                                                      
강남구  3238  150.619195  561052  556164   4888  65060  0.871220  11.596073
강동구  1010  166.490765  440359  436223   4136  56161  0.939234  12.753458
강북구   831  125.203252  328002  324479   3523  56530  1.074079  17.234651
강서구   911  134.793814  608255  601691   6564  76032  1.079153  12.500021
관악구  2109  149.290780  520929  503297  17632  70046  3.384722  13.446362

#### 🔰 상관계수 

- `corr()`
- correlation의 약자이다.

- 데이터의 관계를 찾을 때, 최소한의 근거가 있어야 해당 데이터를 비교하는 의미가 있다.
- 상관계수가 0.2 이상인 데이터를 비교하는 것은 의미가 있다.
- 상관계수 연상을 위해서는 int나 float 자료형이어야 한다.
- value가 object에 해당하는 string일 경우 상관계수 연산은 불가능하다.

In [12]:
merge_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 25 entries, 강남구 to 중랑구
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   소계      25 non-null     int64  
 1   최근증가율   25 non-null     float64
 2   인구수     25 non-null     int64  
 3   한국인     25 non-null     int64  
 4   외국인     25 non-null     int64  
 5   고령자     25 non-null     int64  
 6   외국인비율   25 non-null     float64
 7   고령자비율   25 non-null     float64
dtypes: float64(3), int64(5)
memory usage: 1.8+ KB


In [13]:
merge_data.corr()

소계     최근증가율       인구수       한국인       외국인       고령자     외국인비율  \
소계     1.000000 -0.264378  0.232555  0.227852  0.030421  0.163905 -0.045956   
최근증가율 -0.264378  1.000000 -0.097165 -0.086341 -0.156421 -0.072251 -0.047102   
인구수    0.232555 -0.097165  1.000000  0.998151 -0.167243  0.936737 -0.601076   
한국인    0.227852 -0.086341  0.998151  1.000000 -0.226853  0.936155 -0.645463   
외국인    0.030421 -0.156421 -0.167243 -0.226853  1.000000 -0.175318  0.838612   
고령자    0.163905 -0.072251  0.936737  0.936155 -0.175318  1.000000 -0.620300   
외국인비율 -0.045956 -0.047102 -0.601076 -0.645463  0.838612 -0.620300  1.000000   
고령자비율 -0.267841  0.190396 -0.637414 -0.628360 -0.021147 -0.348840  0.242816   

          고령자비율  
소계    -0.267841  
최근증가율  0.190396  
인구수   -0.637414  
한국인   -0.628360  
외국인   -0.021147  
고령자   -0.348840  
외국인비율  0.242816  
고령자비율  1.000000

- CCTV 전체 수(소계)와 가장 상관관계가 있는 데이터는 "인구수"(0.232555)이다.

- 인구대비 CCTV 비율을 새로운 컬럼으로 만든다.

In [14]:
merge_data["CCTV비율"] = merge_data["소계"] / merge_data["인구수"] * 100
merge_data.head()

소계       최근증가율     인구수     한국인    외국인    고령자     외국인비율      고령자비율  \
구별                                                                         
강남구  3238  150.619195  561052  556164   4888  65060  0.871220  11.596073   
강동구  1010  166.490765  440359  436223   4136  56161  0.939234  12.753458   
강북구   831  125.203252  328002  324479   3523  56530  1.074079  17.234651   
강서구   911  134.793814  608255  601691   6564  76032  1.079153  12.500021   
관악구  2109  149.290780  520929  503297  17632  70046  3.384722  13.446362   

       CCTV비율  
구별             
강남구  0.577130  
강동구  0.229358  
강북구  0.253352  
강서구  0.149773  
관악구  0.404854

- 인구대비 CCTV 비율이 높은 구는?

In [15]:
merge_data.sort_values(by="CCTV비율", ascending=False).head()

소계       최근증가율     인구수     한국인    외국인    고령자     외국인비율      고령자비율  \
구별                                                                         
종로구  1619  248.922414  164257  154770   9487  26182  5.775705  15.939656   
용산구  2096   53.216374  244444  229161  15283  36882  6.252148  15.088118   
중구   1023  147.699758  134593  125709   8884  21384  6.600640  15.887899   
강남구  3238  150.619195  561052  556164   4888  65060  0.871220  11.596073   
금천구  1348  100.000000  253491  235154  18337  34170  7.233787  13.479769   

       CCTV비율  
구별             
종로구  0.985651  
용산구  0.857456  
중구   0.760069  
강남구  0.577130  
금천구  0.531774

- 인구대비 CCTV 비율이 낮은 구는?

In [16]:
merge_data.sort_values(["CCTV비율"]).head()

소계       최근증가율     인구수     한국인    외국인    고령자     외국인비율      고령자비율  \
구별                                                                         
강서구   911  134.793814  608255  601691   6564  76032  1.079153  12.500021   
송파구  1081  104.347826  671173  664496   6677  76582  0.994825  11.410173   
중랑구   916   79.960707  412780  408226   4554  59262  1.103251  14.356800   
강동구  1010  166.490765  440359  436223   4136  56161  0.939234  12.753458   
광진구   878   53.228621  372298  357703  14595  43953  3.920247  11.805865   

       CCTV비율  
구별             
강서구  0.149773  
송파구  0.161061  
중랑구  0.221910  
강동구  0.229358  
광진구  0.235833

In [18]:
merge_data.to_excel("./result_data/03_merge_data_result.xlsx")